data_model/src/lib.rs

```diff
@@ -281,6 +245,32 @@ pub mod parameter {
     pub mod model {
         use super::*;
 
+        #[derive(
+            Debug,
+            Clone,
+            PartialEq,
+            Eq,
+            PartialOrd,
+            Ord,
+            FromVariant,
+            Decode,
+            Encode,
+            Deserialize,
+            Serialize,
+            IntoSchema,
+        )]
+        #[ffi_type(local)]
+        pub enum ParameterValueBox {
+            TransactionLimits(transaction::TransactionLimits),
+            MetadataLimits(metadata::Limits),
+            LengthLimits(LengthLimits),
+            Numeric(
+                #[skip_from]
+                #[skip_try_from]
+                Numeric,
+            ),
+        }
```

Comment:
1. I think that we might want to split `***Limits` into it's constituents
2. I think that having `Numeric` is sufficient because values should be loosely typed and checked in executor

Given the previous 2 points, I think that instead of `ParameterValueBox` we could just have `StringWithJson` as we do for permission tokens. This is because the value for parameters defined in executor can indeed be anything. Even though there are some hard coded, I don't think it's necessarily worth to have them strongly typed inside an enum

In any case the design here should be reconsidered

Reply:
Currently values for `Parameter` used not by executor, but used by `wsv` and `sumeragi`. 
So i think having strong typing is beneficial.
But i don't like current design of `Parameter` and `SetParameter` isi. 
Imo it should be singular instruction for setting on-chain configuration of iroha.
I think we already have some issue about that.